### PR TITLE
feat: improved boilerplate package

### PR DESCRIPTION
**Pull request by thampe**
- [x] I'm confirming that if i made any changes to public APIs or exposed any public APIs they are doccumented.

Hi,

I wanted to take a look at vuestorefront next. Initially I wanted to start a new api-client implementation, but the boilerplate package did not seem to work. So I updated the boilerplate package. At first it seemed like an easy/tiny fix to make it work, but there were some things missing in the boilerplate package and now I added them. Now the boilerplate package at least runs without significant errors with `yarn dev:bp`.

Could someone take a look at it? I am by no means an expert in this field, I mostly do php/magento stuff. 

Thanks and greetings
\- Thomas

### CODE REVIEW

**Comment by rmakara** [comment]
Thank you @thampe :) we will take a look at that.

File: packages/boilerplate/api-client/src/api/getCategory/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {Category} from '../../types';
+
+export default async function (): Promise<Category[]> {
```
Comment (filrak) [inline packages/boilerplate/api-client/src/api/getCategory/index.ts:4]: Have you seen [this](https://vsf-next-docs.netlify.com/integration/api-client.html) doc? Would be cool to have an example including overriding mechanism.

File: packages/boilerplate/api-client/src/index.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
 import getProduct from './api/getProduct';
+import getCategory from './api/getCategory';
 import { apiClientFactory } from '@vue-storefront/factories';
 
-const { setup, override } = apiClientFactory<any, any>();
+const { setup, override } = apiClientFactory<any, any>({
+  defaultSettings: {},
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  onSetup: (config: any) => {}
+});
```
Comment (filrak) [inline packages/boilerplate/api-client/src/index.ts:11]: how about providing settings for api client methods as well?
I mean something like

```
const settings = getSettings()
```

File: packages/boilerplate/theme/nuxt.config.js [hunk 3]
```diff
@@ -8,25 +13,37 @@ export default {
...
+  ],
+  buildModules: [
+    // to core
+    '@nuxt/typescript-build',
     ['@vue-storefront/nuxt', {
       coreDevelopment: true,
       useRawSource: {
         dev: [
-          '@vue-storefront/boilerplate-composables'
+          '@vue-storefront/boilerplate-composables',
```
Comment (filrak) [inline packages/boilerplate/theme/nuxt.config.js]: the new convention for composables package naming is without `composables` part (eg. `@vue-storefront/commercetools`)

File: packages/boilerplate/theme/plugins/boilerplate-config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+export const config = {
+  api: {},
+  locale: 'en',
```
Comment (filrak) [inline packages/boilerplate/theme/plugins/boilerplate-config.js]: All those parts are integration-specific. For example - not every integration will use cookies. I think its best to keep the boilerplate config completely clean 9and also get rid of cookie universal pckg).

File: packages/boilerplate/theme/plugins/boilerplate.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
 import { setup } from '@vue-storefront/boilerplate-api';
+import { config } from './boilerplate-config';
```
Comment (filrak) [inline packages/boilerplate/theme/plugins/boilerplate.js]: How about keeping it in one file (config in `boilerplate.js`) or making a folder to group those files? It doesn't seem right for me to have a file that is not a plugin in `plugins` dir (i know we have it in CT integration - I'm also thinking about a better way to structure this)

**Review by filrak — COMMENTED** [review]
Hey! 

This is absolutely amazing piece of work and will help us a lot in upcoming integrations! Thank you so much!

image

I left few suggestions on how we can improve the boilerplate and give people even a better base. Please let me know what you think about them :)

Also please don't get scared of conflicts. We just changed one path so it should be straightforward to fix that ( `utils` `interfaces` and `factories` were merged to `@vue-storefront/core`(

Regarding changes - we can group them into a new issue to merge the PR faster so people can already apply new APIs to boilerplate as well in their PRs, wdyt?

File: packages/boilerplate/api-client/src/api/getCategory/index.ts (hunk 1 above)
Comment (thampe) [reply packages/boilerplate/api-client/src/api/getCategory/index.ts:4]: No Problem. I have added the overrides to the example api implementations.

File: packages/boilerplate/api-client/src/index.ts (hunk 2 above)
Comment (thampe) [reply packages/boilerplate/api-client/src/index.ts:11]: 👍

File: packages/boilerplate/theme/nuxt.config.js (hunk 3 above)
Comment (thampe) [reply packages/boilerplate/theme/nuxt.config.js]: 👍

File: packages/boilerplate/theme/plugins/boilerplate-config.js (hunk 4 above)
Comment (thampe) [reply packages/boilerplate/theme/plugins/boilerplate-config.js]: Ok. I have removed the config entirely.

File: packages/boilerplate/theme/plugins/boilerplate.js (hunk 5 above)
Comment (thampe) [reply packages/boilerplate/theme/plugins/boilerplate.js]: 👍

**Pull request merged**